lang/csharp/src/apache/test/Reflect/TestFromAvroProject.cs

```diff
@@ -264,6 +266,15 @@ private void DoAssertions(Z z, Z zz)
                 Assert.IsNotNull(zz.myA);
                 Assert.AreEqual(z.myA.f1, zz.myA.f1);
             }
+            if (z.myNullableA == null)
```

Comment:
These tests pass, even before the code changes in `ClassCache.cs`. I think it might be because `A` is already cached via the `myA` field.

Try creating a new class that is only used by this one field so you can exercise this functionality. Make a note in the documentation on that class that we shouldn't use it in other fields in the future for other test cases.

Reply:
Added new test